Essentials/src/main/java/com/earth2me/essentials/commands/Commandvanish.java

```diff
@@ -40,6 +41,19 @@ void togglePlayer(final CommandSource sender, final User user, Boolean enabled)
         if (enabled) {
             user.sendMessage(tl("vanished"));
         }
+
+        final Player player = user.getBase();
+        final boolean vanishBroadcast = ess.getSettings().vanishBroadcast();
+        if (vanishBroadcast) {
+            String broadcastMessage = "";
+            if (enabled) {
+                broadcastMessage = tl("vanishBroadcast", player.getName(), player.getDisplayName());
+            } else {
+                broadcastMessage = tl("unvanishBroadcast", player.getName(), player.getDisplayName());
```

Comment:
Should use the actual join/quit message here, not hard-coded to tl, as other plugins will modify it

Reply:
Is it possible tho? Plugins can modify messages only on firing of specific event, but firing join/quit event may lead to huge problems. The only way I see is to add the message into `VanishStatusChangeEvent` and hope that other plugins/developers with hook into it.